src/js/gpii-pouchdb-lucene.js

```diff
@@ -0,0 +1,253 @@
+    fluid.each(libJarSegments, function(segment){
+        classpathSegments.push(path.join("lib", segment));
+    });
+    var classpathString = classpathSegments.join(";");
+
+    return "java -Xmx1g -Did=\""+ that.id + "\" -cp " + classpathString + " com.github.rnewson.couchdb.lucene.Main";
+};
+
+//gpii.pouch.lucene.getJavaPath = function () {
+//    return path.join(process.env.JAVA_HOME, "bin", "java");
+//};
+
+gpii.pouch.lucene.init = function (that){
+    // Use our ID as a unique identifier so that we can avoid clobbering another instance.
+    var outputDir = path.resolve(that.options.tmpDir, that.id);
+
+    // Unpack the "dist" zip created when we were installed to create a new working directory.
+    fluid.log("Creating working directory for this instance...");
+    fluid.log("output dir = " + outputDir);
+
+    if (!fs.existsSync(that.options.zipPath)) {
+        fluid.fail("The specified couchdb-lucene zip file doesn't exist: '" + that.options.zipPath + "'...");
+    }
+
+    var distZip = new AdmZip(that.options.zipPath);
+    distZip.extractAllTo(outputDir);
+
+    that.workingDir = path.resolve(outputDir, "couchdb-lucene-" + npmSettings.config.version);
+    var confFile   = path.resolve(that.workingDir, "conf", "couchdb-lucene.ini");
+
+    fluid.log("Generating configuration file...");
+    var iniContent = gpii.pouch.lucene.generateIniContent(that);
+    fs.writeFileSync(confFile, iniContent);
+
+    // Start the service using either the batch file on windows, or the shell script on anything else.
+    fluid.log("Starting couchdb-lucene...");
+
+    var isWindows = os.platform().indexOf("win") === 0;
+
+    var shell     = isWindows ? "cmd.exe" : "sh";
+    var args      = isWindows ? ["/c", gpii.pouch.lucene.generateWindowsCommandString(that)] : [path.resolve(that.workingDir, "bin/run")];
+
+    that.process = child_process.spawn(shell, args, { cwd: that.workingDir });
+    that.process.stdout.on("data", that.checkForStartupMessage); // Watch for "accepting connections", which means startup is complete
+
+    // Display the child processes log messages on stdout/stderr.
+    //
+    // We have to pipe the streams created by the child process to their equivalents in `process` because
+    // `process.stdout` and `process.stderr` are write only streams, and cannot be monitored for changes.
+    //
+    // Otherwise we would simply use the `inherit` option when we spawn the process.
+    that.process.stdout.pipe(process.stdout);
+    that.process.stderr.pipe(process.stderr);
+
+    that.process.on("close", that.events.onShutdownComplete.fire);
+};
+
+// Ensure that the service is stopped on component destruction
+gpii.pouch.lucene.stopProcess = function (that) {
+    if (that.process) {
+        // Hello, cruel Windows.
+        if (os.platform().indexOf("win") === 0) {
+            var pidLookupCommand = "wmic process where \"name='java.exe' and commandline like '%" + that.id + "%'\" get processid";
```

Comment:
Please package this as a reusable function accepting the variable material (process name and commandline tag) so that it's ready to be used more generally.